elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchHSQueryImpl.java

```diff
@@ -540,23 +541,17 @@ EntityInfo convertQueryHit(JsonObject hit) {
 							}
 							break;
 						case ProjectionConstants.THIS:
-							indexesOfThis.add( i );
+							// Use THIS as placeholder. It will be replaced
+							// when we populate the EntityInfo with the real entity.
+							projections[i] = ProjectionConstants.THIS;
```

Comment:
Rather than reusing the String from `ProjectionConstants.THIS`, I was considering using a marker object.

Would you prefer to create a (static final) singleton having just a nice toString() implementation so it doesn't get too confusing during debug?
You would still use reference equality to check for the marker, but it makes me more comfortable than doing it on a string.